Fix 'defaults' option in the nxos_config module

OlivierB opened the pull request:
##### SUMMARY
Nxos get_config is allways called with the 'all' option.
* Fix flag's calculation
* Add tests

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
nxos_config

##### ADDITIONAL INFORMATION

* This issue seems to be added in the PR #26565.
* The eos_config module has already been fixed:
  * #26798 logic fix.
  * #42300 functional change: 'all' option is used on the backup config.

I have not changed the backup behavior as in the eos_config module.

OlivierB committed:
Fix 'defaults' option in the nxos_config module

Nxos get_config is allways called with the 'all' option.
* Fix flag's calculation
* Add tests

trishnaguha commented on lib/ansible/modules/network/nxos/nxos_config.py:
```diff
@@ -290,9 +290,12 @@ def get_running_config(module, config=None):
     contents = module.params['running_config']
     if not contents:
         if not module.params['defaults'] and config:
+            # Optimization: The running-config has already been recovered by the backup
```
To make it work, `not module.params['defaults']` needs to be removed from this condition.

trishnaguha reviewed (commented):
@OlivierB Also instead of only fixing it in `get_running_config`, I would something like https://github.com/trishnaguha/ansible/commit/4f343491a3de63cbc9fd99eb7c120a34f7c143aa

OlivierB commented:
@trishnaguha OK !
I will implement it like in the eos_config module.

OlivierB committed:
nxos_config: the 'backup' option take into account the value of 'defaults' option

If 'defaults' option is true, the running-config backup is done with the all
keyword.

trishnaguha reviewed (approved):
Thanks for the PR!
